ui/src/Components/EditProfile.jsx

```diff
@@ -0,0 +1,266 @@
+    checkIfBlank(){
+        if(this.state.firstName == ""){
+            this.state.firstName = this.state.profile.result.first_name;
+        }  
+        if(this.state.lastName  == ""){
+            this.state.lastName = this.state.profile.result.last_name;
+        }    
+        if(this.state.number  == ""){
+            this.state.number = this.state.profile.result.contact_num;         
+        }     
+        if(this.state.number  == ""){
+            this.state.email = this.state.profile.result.email; 
+        }  
+    };
+
+    /** NEW API to be set - PUT request that updates the Entry!*/
+    handleSubmit(event) {
+        this.checkIfBlank();
+        event.preventDefault();
+        fetch(URL,
+            {
+              method: "PUT",
+              headers: {
+                "Accept": "application/json",
+                "Content-Type": "application/json"
+                },
+              body: JSON.stringify( { 
+                firstName:this.state.firstName, 
+                lastName:this.state.lastName,
+                email:this.state.email, 
+                number:this.state.number, 
+                reference1Name:this.state.ref1Name,
+                reference1Email:this.state.ref1Email,
+                reference2Name:this.state.ref2Name,
+                reference2Email:this.state.ref2Email,
+                govId:this.state.govIdFlag,
+                toolPic:this.state.toolPicFlag,
+                id:this.state.props.match.params.id, //added profile ID to payload
+                }),
+            })
+            .then(res => res.json())
+            .then(res => {
+                if (res.govIdUrl!=='') { //upload gov id
+                    fetch(res.govIdUrl, {
+                        method: "PUT",
+                        body: this.state.govId,
+                        headers: {
+                            "Content-Type": "application/pdf",
+                            "x-amx-acl" : "public-read",
+                        },
+                    })
+                    .then((res) => {
+                        console.log(res);
+                    });
+                }
+                if (res.toolPicUrl!=='') { //upload tool file
+                    fetch(res.toolPicUrl, {
+                        method: "PUT",
+                        body: this.state.toolPic,
+                        headers: {
+                            "Content-Type": "application/pdf",
+                            "x-amx-acl" : "public-read",
+                        },
+                    })
+                }
+            window.location.href = ('/profiles/' + res.result) //redirect to Profile View Page
+            })
+            .catch(error => console.error('Error:', error));
+    };
+    
+    render() {
+        return (
+            <>
+            <NavBar/>
+            {this.state.loading === (
+            <div>Loading...</div>
+            )}
```

Comment:
I don't think that this loading logic works properly bro. No matter what the Profile Does not Exist message will be displayed. What you can do is do smt like this:
```
render() {
if (this.state.loading) {
return(
<div><h2>Loading...</h2></div>
)
} else {
return(
//everything else after this
)
}

}
```

Also, instead of the message  "loading...", we could also leverage the brand new loading component that Matt made for his story, it looks p nice.

Reply:
So this is a semantic error in all our viewProfile component 
It is also because the "loading" variable is not updated after being initialized as well. I would say lets get rid of it and provide a better "loading" experience for Sprint 4 (even though its not that big of a priority imo). For an MVP its functional. 
I am getting rid of the loading, and we should open an issue for the viewProfile component so we can track it

Reply:
Wait sorry bro, I don't understand. On line 53 after we fetch the data, we do update the loading state to be false, that would kick off a re-render and make the code snippet above work. I think I may be misunderstanding your comment or maybe I wasn't clear w my statement above.

Reply:
I updated with a new commit - plz see the video I sent you offline